trustfall_core/src/schema/adapter/mod.rs

```diff
@@ -214,15 +215,34 @@ impl<'a> crate::interpreter::Adapter<'a> for SchemaAdapter<'a> {
         &self,
         edge_name: &Arc<str>,
         _parameters: &EdgeParameters,
-        _resolve_info: &ResolveInfo,
+        resolve_info: &ResolveInfo,
     ) -> VertexIterator<'a, Self::Vertex> {
         match edge_name.as_ref() {
             "VertexType" => {
                 let root_query_type = self.schema.query_type_name();
-                Box::new(self.schema.vertex_types.values().filter_map(move |v| {
-                    (v.name.node != root_query_type)
-                        .then(|| SchemaVertex::VertexType(VertexType::new(v)))
-                }))
+
+                if let Some(crate::interpreter::CandidateValue::Single(FieldValue::String(str))) =
```

Comment:
Can we handle `CandidateValue::Multiple` here?

Otherwise queries with `name @filter(op: "=", ...)` will be fast, but queries with `name @filter(op: "one_of", ...)` will be slow.

Reply:
Done!